src/daemon/rpmostreed-deployment-utils.c

```diff
@@ -111,31 +111,42 @@ rpmostreed_deployment_get_for_index (OstreeSysroot *sysroot,
 }
 
 static gboolean
-variant_add_gpg_results (OstreeRepo  *repo,
-                         const gchar *origin_refspec,
-                         const gchar *checksum,
-                         GVariantDict *dict,
-                         GError     **error)
+variant_add_origin_status (OstreeRepo  *repo,
+                           const gchar *origin_refspec,
+                           const gchar *checksum,
+                           GVariantDict *dict,
+                           GError     **error)
 {
-  GLNX_AUTO_PREFIX_ERROR ("GPG verification error", error);
+  GLNX_AUTO_PREFIX_ERROR ("Loading origin status", error);
 
   g_autofree gchar *remote = NULL;
   if (!ostree_parse_refspec (origin_refspec, &remote, NULL, error))
     return FALSE;
 
+  g_autoptr(GError) local_error = NULL;
   gboolean gpg_verify = FALSE;
   if (remote)
     {
-      if (!ostree_repo_remote_get_gpg_verify (repo, remote, &gpg_verify, error))
-        return FALSE;
+      if (!ostree_repo_remote_get_gpg_verify (repo, remote, &gpg_verify, &local_error))
+        {
+          /* If the remote doesn't exist, let's note that so that status can
+           * render it specially.
+           */
+          if (g_error_matches (local_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+            {
+              g_variant_dict_insert (dict, "remote-not-found", "s", local_error->message);
```

Comment:
Hmm, it seems odd to me to proxy back the exact error message. We already know it's `NOT_FOUND`, right? Let's just make this a `b`?

Reply:
I actually had that originally and changed it; it felt easier to reuse the existing error message.  I also had further thoughts that we could start to show e.g. `unconfigured-state` this way.

Reply:
In that case, how about making the key also generic to future-proof it, e.g. `remote-error`?

Reply:
Sure, done :arrow_down: